resources/tentative/todomvc-react-18/src/components/App.tsx

```diff
@@ -0,0 +1,94 @@
+import { RouterProvider, createHashRouter } from "react-router-dom";
+import { TodoListPage } from "./TodoListPage";
+import { loader } from "../logic/todo-loader";
+import { pushTodo, toggleTodo, toggleAll, removeTodo, removeCompleted, setText, populate } from "../logic/todo-model";
+
+const router = createHashRouter([
+    {
+        path: "/:filter?",
+        element: <TodoListPage />,
+        loader,
+    },
+    {
+        path: "/api/todos",
+        children: [
+            {
+                path: "new",
+                action: async ({ request }) => {
```

Comment:
oh, I like the way the router is set up

Reply:
Thanks!
Yeah, it's the new way introduced in react-router v6.4, I like it as well!